### PR TITLE
UX: Header user/profile drop-down doesn't automatically close

### PR DESCRIPTION
**Pull request by partpa**
#### What's this PR do? Any additional context?
-      It will close drop down automatically when you will click on screen anywhere other than drop-down.

#### Where should the reviewer start?
-      SignIn Production site and open drop-down. Try to close it by clicking outside of drop-down.

#### How can this be manually tested? (brief repro steps)
-     Before : drop-down doesn't automatically close when click outside on screen.
       After: drop-down does automatically close when click outside on screen.

#### What are the relevant TFS items? (list id numbers)
-       Bug:  84810

#### Definition of done:
- [x] Title of the pull request is clear and informative
- [x] Add pull request hyperlink to relevant TFS items
- [ ] For large or complex change: schedule an in-person review session

[//]: # ( todo: Is there appropriate test coverage? )
[//]: # ( todo: Does this PR require a new Selenium test? )
[//]: # ( todo: Is there appropriate logging/monitoring included? )

#### Reminders BEFORE merging
1. Get at least two approvals
1. If you're merging into the development branch then "flatten" or "squash" commits
1. If merging from development into master then don't "flatten" or "squash" commits

#### Reminders AFTER merging
1. Delete the remote branch
1. Resolve relevant TFS items
1. (reverse merge) If you merged into master then check to see if there are any changes in master that can be merged down to the development branch (like hotfixes, etc)

[//]: # ( todo: If you merged into development branch then verify change in our "rolling deployment" environment. Then notify stakeholders interested in or involved with the change )

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 3 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 4 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 5 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

### CODE REVIEW

File: lms/static/js/my_courses_dropdown.js [hunk 1]
```diff
@@ -90,4 +90,11 @@ $(document).ready(function() {
     $dropdownMenu.on('keydown', function(event) {
         catchKeyPress($(this), event);
     });
+	$(this).on('click', function(){
```
Comment (sdolenc) [inline lms/static/js/my_courses_dropdown.js]: what if the click is within the dropdown menu itself? it shouldn't dismiss in that case.

File: lms/static/js/my_courses_dropdown.js [hunk 2]
```diff
@@ -90,4 +90,11 @@ $(document).ready(function() {
...
+        if($dropdownMenu.hasClass('expanded')){
+                $dropdownMenu.removeClass('expanded');
+                $dropdownMenuToggle.removeClass('active').attr('aria-expanded', 'false').focus();
+        }
```
Comment (sdolenc) [inline lms/static/js/my_courses_dropdown.js]: is this the standard way to do this? I feel like there might be a more elegant approach. for example, the customer presses "the escape" key will also want to dismiss the flyout

Comment (sdolenc) [inline lms/static/js/my_courses_dropdown.js]: it looks fairly close to a popular strategy on stackoverflow, but even these answers are missing the "escape" key
https://stackoverflow.com/questions/2868582/click-outside-menu-to-close-in-jquery

File: lms/static/js/my_courses_dropdown.js (hunk 1 above)
Comment (sdolenc) [inline lms/static/js/my_courses_dropdown.js]: this is the issue i'm blocking on btw.

**Comment by partpa** [comment]
- I will fix clicking within dropdown  and standardize style.
- For 'esc' key, it is working when dropdown button is on fucus.

**Pull request merged**